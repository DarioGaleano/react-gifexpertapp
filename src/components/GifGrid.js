import React from 'react';
// import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	console.log({ category });
	const { data, loading } = useFetchGifs(category);

	return (
		<>
			<h3 className='animate__animated animate__fadeIn'>{category}</h3>
			{loading && <p className='animate__animated animate__flash'>Loading...</p>}
			<div className='card-grid'>
				{data.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</>
	);
};

GifGrid.propTypes = {};
