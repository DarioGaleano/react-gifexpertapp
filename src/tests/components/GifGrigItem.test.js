import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test on <GifGridItem />', () => {
	const title = 'A title';
	const url = 'http://localhost.algo/ssk.jpg';

	let wrapper = shallow(<GifGridItem title={title} url={url} />);
	beforeEach(() => {
		wrapper = shallow(<GifGridItem title={title} url={url} />);
	});

	test('should show the component correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should have a paragraph with the title', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('should have an image whit the url and alt props', () => {
		const img = wrapper.find('img');
		expect(img.prop('src')).toBe(url);
		expect(img.prop('alt')).toBe(title);
	});

	test('should have a div container with the class anitame__fadeIn', () => {
		const div = wrapper.find('div');
		const className = div.prop('className');
		expect(className.includes('animate__fadeIn')).toBe(true);
	});
});
