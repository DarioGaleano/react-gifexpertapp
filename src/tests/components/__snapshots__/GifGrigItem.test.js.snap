// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem /> should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="http://localhost.algo/ssk.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
